### ISSUE TITLE
[UX] Layout/block UI: Better field block summaries.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3236]
## Describe your issue or idea

As it is currently, the description of any field block added to the layout has "Displays values of `[field_name]` field" as its description:

<img width="497" alt="screen shot 2018-08-04 at 6 05 30 am" src="https://user-images.githubusercontent.com/2423362/43663542-3cbf1894-97ad-11e8-844f-00f689bc1481.png">

This does not give any other useful information, and in fact using plural "values" is misleading when the field has cardinality 1.

---
PR by @klonos: https://github.com/backdrop/backdrop/pull/2261

**Comment by klonos** [comment]
...so the current PR does the following:

- Detects cardinality of the field of the block, and says "value"/"values" accordingly (using `format_plural()`).
- Adds "the" in front of "value"/"values", and in front of the field name for a more natural language.
- Removes the double quotes from the field name (using `%` within `t()`/`format_plural()` already renders the string in italics, which is sufficient).
- Appends the field type human-readable label to the description.

Here's what it looks like:

<img width="495" alt="screen shot 2018-08-04 at 6 03 44 am" src="https://user-images.githubusercontent.com/2423362/43663852-40415968-97ae-11e8-96dc-6c7e8ea31614.png">

**Comment by opi** [comment]
The double parenthesis looks strange...

**Comment by klonos** [comment]
> The double parenthesis looks strange...

I know, but it is coming from the default field type description in https://github.com/backdrop/backdrop/blob/1.x/core/modules/field/modules/text/text.module#L38. What do you propose we do instead?

Comment by @opi on Gitter:

> what about removing the "Displays the" prefix ? seems pretty useless to me

I was going for natural language. I see now that some block descriptions use statements like "The default content of this page." or "The current page breadcrumb trail." or "A hierarchical list of links..." etc., but we then also have descriptions that use proper phrases, starting with verbs, like "Contains elements typical of...". and "Displays a link to ...". I don't mind either; I was just keeping what was already there.

**Comment by opi** [comment]
OK for this *natural language* thing, as I'm not that fluent in english, my opinion doesn't worth so much.

For the field type display, I made a test to print it in the summary part: ![](https://i.imgur.com/aClzKLs.png)

**Comment by klonos** [comment]
I was going for a simple PR/change, while adding setting labels in the summary will mean updating the `_field_formatter_settings_summary` hooks for all field types. Printing the field type along with the summary is not a bad idea though @opi 👍...here's how this would look with the labels of settings bolded (for better readability):

<img width="579" alt="screen shot 2018-08-05 at 6 52 05 am" src="https://user-images.githubusercontent.com/2423362/43680391-fbd6dbb6-987c-11e8-9ffa-ba4be2500d98.png">

I will file a separate PR so people can decide.

**Comment by opi** [comment]
Updating `hook_field_formatter_settings_summary` will also affect the content type display setting page, which is not what we want (maybe I'm wrong). ![](https://i.imgur.com/tGStqVM.png)
Field type could be added into summary from the `FieldBlock->getAdminPreview()` method.

Also, bolding the label is nice, but will affect all existing translations... but why not.

**Comment by klonos** [comment]
> Updating `hook_field_formatter_settings_summary` will also affect the content type display setting page ... Also, bolding the label is nice, but will affect all existing translations... but why not.

Yeah, I have thought of that, but I like uniformity/consistency, so I like how this looks:

<img width="1176" alt="screen shot 2018-08-06 at 5 45 32 am" src="https://user-images.githubusercontent.com/2423362/43689328-2800eb54-993c-11e8-8a24-2cffa38c3c3a.png">

I would like to get some feedback from others too before spending time on a new PR.

**Comment by herbdool** [comment]
I'll just comment on one part: no need to insert "the". It sounds just as natural with the "the" because of the context. It's not unusual to omit the word on signs and messages since the space is limited and need to convey info quickly.

**Comment by opi** [comment]
I do not filed another PR. My screenshot was just a quick'n'dirty local test.

**Comment by klonos** [comment]
@quicksketch so just to clarify the "needs work" bit, from the comments above I am assuming the following:

1. no bold labels for properties
2. remove the "Displays the" prefix

...anything else?
